bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClientFactory.java

```diff
@@ -56,11 +56,17 @@ public class BigQueryClientFactory implements Serializable {
   private static final Map<BigQueryClientFactory, BigQueryWriteClient> writeClientMap =
       new HashMap<>();
 
-  private final Credentials credentials;
   // using the user agent as HeaderProvider is not serializable
   private final HeaderProvider headerProvider;
   private final BigQueryConfig bqConfig;
+
+  // GoogleCredentials are not compatible with Kryo serialization, so we serialize and deserialize
+  // when needed
+  private final byte[] serializedCredentials;
+  private transient volatile Credentials credentials;
+
   private int cachedHashCode = 0;
+  private IdentityTokenSupplier identityTokenSupplier;
```

Comment:
Why is this added?

Reply:
It was in the original PR,  I've added it so it would compile. They class doesn't use it though.